src/realm/query_engine.hpp

```diff
@@ -449,6 +449,20 @@ static size_t find_first_haystack(LeafType& leaf, NeedleContainer& needles, size
 {
     // for a small number of conditions, it is faster to do a linear search than to compute the hash
     // the exact thresholds were found experimentally
+
+    if constexpr (std::is_same_v<LeafType, ArrayInteger>) {
```

Comment:
I am not sure if I understood this. Isn't this the case considering the check in the line below?

`if (needles.size() >= linear_search_threshold && leaf.is_compressed()) {` 

What I am trying to do here is to avoid running this:

```
else {
        for (size_t i = start; i < end; ++i) {
            auto element = leaf.get(i);
            if (needles.count(element))
                return i;
        }
    }
```

Maybe I can move the code to the else branch, instead of having it at the top, to make it more clear.